youtube_dl/extractor/common.py

```diff
@@ -600,6 +600,10 @@ def _request_webpage(self, url_or_request, video_id, note=None, errnote=None, fa
 
         See _download_webpage docstring for arguments specification.
         """
+        page_sleep_interval = self._downloader.params.get('page_sleep_interval')
+        page_sleep_interval = 0 if page_sleep_interval is None else float(page_sleep_interval)
+        time.sleep(page_sleep_interval)
```

Comment:
1. DRY. Generalize with download sleep interval.
2. Do not sleep if interval is invalid.
3. There must be min and max intervals for randomization.